azure-pipelines.yml

```diff
@@ -0,0 +1,296 @@
+trigger:
+- master
+
+pool:
+  # need to run on linux pool in order to run docker linux containers
+  vmImage: 'ubuntu-latest'
+
+variables:
+  # smoke tests are run inside of docker linux containers so don't use windows performance counters anyways
+  GradleParams: --info -DskipWinNative=true
+  isRelease: false
+
+jobs:
+- job: smoke
+  strategy:
+    matrix:
```

Comment:
This will get unwieldy as we add dimensions. I'd like to generate the matrix.

A simple example: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#multi-configuration (2nd one)

Since it just needs json, a gradle task might be easier.

Reply:
That's cool that you can generate the matrix, I was wondering about that! Let's come back to it later? (also, I tried splitting out JRE as a 3rd dimension but it ended up exceeding 256 matrix slice limit 😭)

Reply:
Sounds good.

Maybe we can split this into two builds. I saw you can specify the yaml file to use in the build definition settings.